tests/std/tests/P1222R4_flat_set/test.cpp

```diff
@@ -73,6 +73,22 @@ void assert_reversible_container_requirements(const T& s) {
     static_assert(is_convertible_v<typename T::reverse_iterator, typename T::const_reverse_iterator>);
 }
 
+template <class T>
+void assert_set_requirements() {
+    using iterator       = T::iterator;
+    using const_iterator = T::const_iterator;
+    using key_type       = T::key_type;
+    using value_type     = T::value_type;
+
+    static_assert(_Constant_iterator<const_iterator>);

```

Comment:
We could avoid directly using internal machinery here by instead:
```suggestion
    static_assert(same_as<std::const_iterator<const_iterator>, const_iterator>);
```
(and similarly on line 88)